drivers/adc/adc_stm32.c

```diff
@@ -1646,55 +1661,38 @@ static int adc_stm32_init(const struct device *dev)
 	}
 #endif
 
-#if defined(CONFIG_SOC_SERIES_STM32L4X) || \
-	defined(CONFIG_SOC_SERIES_STM32L5X) || \
-	defined(CONFIG_SOC_SERIES_STM32WBX) || \
-	defined(CONFIG_SOC_SERIES_STM32G4X) || \
-	defined(CONFIG_SOC_SERIES_STM32H5X) || \
-	defined(CONFIG_SOC_SERIES_STM32H7X) || \
-	defined(CONFIG_SOC_SERIES_STM32H7RSX) || \
-	defined(CONFIG_SOC_SERIES_STM32N6X) || \
-	defined(CONFIG_SOC_SERIES_STM32U3X) || \
-	defined(CONFIG_SOC_SERIES_STM32U5X)
-	/*
-	 * L4, WB, G4, H5, H7 and U5 series STM32 needs to be awaken from deep sleep
-	 * mode, and restore its calibration parameters if there are some
-	 * previously stored calibration parameters.
-	 */
-	LL_ADC_DisableDeepPowerDown(adc);
+#if ANY_ADC_HAS_DEEP_POWERDOWN()
+	if (config->has_deep_powerdown) {
+		LL_ADC_DisableDeepPowerDown(adc);
+	}
 #endif
 
+#if !ANY_ADC_INTERNAL_REGULATOR_TYPE_IS(INTERNAL_REGULATOR_NONE)
```

Comment:
This is probably incorrect: `NOT(ANY(xyz))` is not equivalent to `ANY(NOT(xyz))`.
Did you mean:
```suggestion
#if ANY_ADC_INTERNAL_REGULATOR_TYPE_IS(INTERNAL_REGULATOR_DELAY) \
    || ANY_ADC_INTERNAL_REGULATOR_TYPE_IS(INTERNAL_REGULATOR_FLAG)
    /* at least one instance has a regulator */
```

Ditto everywhere where you have this check. You may want to use an intermediary macro e.g. `NEED_INTERNAL_REGULATOR_SUPPORT`